Sources/Rendering/OpenGL/ImageMapper/index.js

```diff
@@ -1310,6 +1318,24 @@ function vtkOpenGLImageMapper(publicAPI, model) {
         ptsArray[9] = spatialExt[1];
         ptsArray[10] = spatialExt[3];
         ptsArray[11] = sliceDepth;
+        if (sliceOffset === 0) {
+          // If the sliceOffset is 0, the scalars array is a view of the
+          // original scalars array. As a micro-optimization, we can get the
+          // range from the original scalars array and set the range to the
+          // scalars array.  This means there is no need to re-calculate the
+          // ranges for the scalars array.
+          ranges = [];
+          for (let i = 0; i < numComp; i++) {
+            const [min, max] = imgScalars.getRange(i);
```

Comment:
maybe we could expose a convenient `getRanges()` function

Reply:
Would you like me to include a `getRanges()` api in `DataArray` as part of this API or a separate?

Reply:
yes, it can be done as part of this PR.

Reply:
I have refactored this and created a `getRanges` function in DataArray public api.